server/user/index.js

```diff
@@ -1,18 +1,22 @@
 'use strict';
 
 const { authenticate, logout } = require('../shared/auth');
+const { ACCEPTED } = require('http-status-codes');
 const { authorize } = require('../shared/auth/mw');
 const ctrl = require('./user.controller');
 const { processPagination } = require('../shared/database/pagination');
 const router = require('express').Router();
 const { User } = require('../shared/database');
 
+// Public routes:
 router
-  // Public routes:
   .post('/login', authenticate('local', { setCookie: true }), ctrl.getProfile)
   .post('/forgot-password', ctrl.forgotPassword)
   .post('/reset-password', authenticate('token'), ctrl.resetPassword)
-  // Protected routes:
+  .post('/reset-token-validation', authenticate('token'), (_, res) => res.sendStatus(ACCEPTED));
```

Comment:
would it make sense for this to be nested under the reset-password route (i.e. `/reset-password/token-status` or similar). Add rate limiting.